entity.go

```diff
@@ -396,14 +395,31 @@ func (e *EntityDefinition) IsCompatible(e2 *EntityDefinition) error {
 	for name, colType2 := range colsMap2 {
 		colType1, ok := colsMap1[name]
 		if !ok {
-			return errors.Errorf("the column %s in entity %s but not in entity %s", name, e1.Name, e2.Name)
+			return errors.Errorf("the column %s in old entity %s but not in new entity", name, e2.Name)
 		}
 		if colType1 != colType2 {
 			return errors.Errorf("the type for column %s mismatch: (%v vs %v)", name, colType1, colType2)
 		}
 	}
 
-	// TODO Handle tags and index comparison in the future
+	// Index can only be added, not mutated
+	if len(e2.Indexes) > len(e1.Indexes) {
```

Comment:
ah ok. Can we just change the if statement to say that then? That way it will be clearer what we're checking.